scripts/west_commands/sign.py

```diff
@@ -191,88 +193,161 @@ def sign(self, command, build_dir, bcfg, formats):
-                    "in build directory's ({}) device tree".
-                    format(slot_size, build_dir))
-
-        b = pathlib.Path(build_dir)
-        cache = cmake.CMakeCache.from_build_dir(build_dir)
-        runner_config = cached_runner_config(build_dir, cache)
-
-        # Build a signed .bin
-        if 'bin' in formats and runner_config.bin_file:
+        sign_base = [tool_path, 'sign',
+                     '--version', '0.0.0+0',
+                     '--align', str(align),
+                     '--header-size', str(vtoff),
+                     '--slot-size', str(size)]
+        sign_base.extend(args.tool_args)
+
+        log.banner('signed binaries:')
+        if in_bin:
             out_bin = args.sbin or str(b / 'zephyr' / 'zephyr.signed.bin')
-            log.inf('Generating:', out_bin)
-            sign_bin = (sign_base + args.tool_args +
-                        [runner_config.bin_file, out_bin])
+            sign_bin = sign_base + [in_bin, out_bin]
+            log.inf('bin: {}'.format(out_bin))
             log.dbg(quote_sh_list(sign_bin))
             subprocess.check_call(sign_bin)
-
-        # Build a signed .hex
-        if 'hex' in formats and runner_config.hex_file:
+        if in_hex:
             out_hex = args.shex or str(b / 'zephyr' / 'zephyr.signed.hex')
-            log.inf('Generating:', out_hex)
-            sign_hex = (sign_base + args.tool_args +
-                        [runner_config.hex_file, out_hex])
+            sign_hex = sign_base + [in_hex, out_hex]
+            log.inf('hex: {}'.format(out_hex))
             log.dbg(quote_sh_list(sign_hex))
             subprocess.check_call(sign_hex)
 
+    @staticmethod
+    def find_imgtool(command, args):
+        if args.tool_path:
+            command.check_force(shutil.which(args.tool_path),
+                                '--tool-path {}: not an executable'.
+                                format(args.tool_path))
+            tool_path = args.tool_path
+        else:
+            tool_path = shutil.which('imgtool')
+            if not tool_path:
+                log.die('imgtool not found; either install it',
+                        '(e.g. "pip3 install imgtool") or provide --tool-path')
+        return tool_path
+
     @staticmethod
     def get_cfg(command, bcfg, item):
         try:
             return bcfg[item]
         except KeyError:
             command.check_force(
-                False,
-                "imgtool parameter unknown, build directory has no {} {}".
-                format('device tree define' if item.startswith('DT_') else
-                       'Kconfig option',
-                       item))
+                False, "build .config is missing a {} value".format(item))
             return None
+
+    @staticmethod
+    def edt_flash_node(b, cache):
+        # Get the EDT Node corresponding to the zephyr,flash chosen DT
+        # node.
+
+        # Retrieve the list of devicetree bindings from cache.
+        try:
```

Comment:
Does all of this boilerplate belong somewhere else? we are gonna end up needing to use DT info more and more I assume as time goes by. Maybe in the common files for west extension commands in Zephyr?

Reply:
Probably eventually, but I want to refactor when I have two users, not before.